File: E2ETestResults/CoreSubset/Core/Utility/RecordCast.ts
import { DoubleRectanglePresModel, RectanglePresModel, SizePresModel } from 'TypeDefs';

import { DoubleRectXY, RectXY } from './Rect';

import { Size } from './Size';

/**
 * Use this class to treat records (basic property bag objects) like other records.
 * You want to do so because the records provide the same properties/interface.
 * Does not make copies or use memory.  (We care less about the function overhead.)
 * Generally, casting to smaller records (sub-sets of properties) is safe, while casting larger is not...
 */
export class RecordCast {
  public static rectPresModelAsRectXY(rpm: RectanglePresModel): RectXY {
    return rpm === null ? null : new RectXY(rpm.x, rpm.y, rpm.w, rpm.h);
  }

  public static doubleRectPresModelAsDoubleRectXY(rpm: DoubleRectanglePresModel): DoubleRectXY {
    return rpm === null ? null : new DoubleRectXY(rpm.doubleLeft, rpm.doubleTop, rpm.width, rpm.height);
  }

  public static sizeAsSizePresModel(sz: Size): SizePresModel {
    return sz === null ? null : new SizePresModel();
  }
}

File: E2ETestResults/CoreSubset/Core/Utility/RecordCast.cs
// ---------------------------------------------------------------------------------------------------------------------
// <copyright file="RecordCast.cs" company="Tableau Software">
//   This file is the copyrighted property of Tableau Software and is protected by registered patents and other
//   applicable U.S. and international laws and regulations.
//
//   Unlicensed use of the contents of this file is prohibited. Please refer to the NOTICES.txt file for further details.
// </copyright>
// ---------------------------------------------------------------------------------------------------------------------

namespace Tableau.JavaScript.Vql.Core
{
    using TypeDefs;

    /// <summary>
    /// Use this class to treat records (basic property bag objects) like other records.
    /// You want to do so because the records provide the same properties/interface.
    /// Does not make copies or use memory.  (We care less about the function overhead.)
    /// Generally, casting to smaller records (sub-sets of properties) is safe, while casting larger is not...
    /// </summary>
    public static class RecordCast
    {
        public static RectXY RectPresModelAsRectXY(RectanglePresModel rpm)
        {
            return rpm == null ? null : new RectXY(rpm.Left, rpm.Top, rpm.Width, rpm.Height);
        }

        public static DoubleRectXY DoubleRectPresModelAsDoubleRectXY(DoubleRectanglePresModel rpm)
        {
            return rpm == null ? null : new DoubleRectXY(rpm.Left, rpm.Top, rpm.Width, rpm.Height);
        }

        public static SizePresModel SizeAsSizePresModel(Size sz)
        {
            return sz == null ? null : new SizePresModel { Width = sz.Width, Height = sz.Height };
        }
    }
}
